build.rs

```diff
@@ -265,6 +280,36 @@ fn build_tor(libevent: Artifacts) {
     println!("cargo:rustc-link-lib=static={}", "tor-dispatch");
     println!("cargo:rustc-link-lib=static={}", "tor-trace");
 
+    if target.contains("windows") {
+        // println!("cargo:rustc-link-search=native=/usr/i686-w64-mingw32/lib");
+
+        // Add the CC's library paths
+        let output = Command::new(format!("{}", compiler.path().display()))
+            .arg("-print-search-dirs")
+            .output()
+            .expect("CC doesn't accept -print-search-dirs");
+
+        let output = std::str::from_utf8(&output.stdout).expect("Invalid output");
+        let lines: Vec<&str> = output
+            .lines()
+            .filter(|line| line.starts_with("libraries: ="))
+            .collect();
+        for line in lines {
+            let equals = line.find("=").unwrap();
+            for path in (&line[(equals + 1)..]).split(':') {
+                println!("cargo:rustc-link-search=native={}", path);
+            }
+        }
```

Comment:
I haven't tried running this yet, but by reading the code i guess what you are trying to do here is:

1. only keep the lines that start with `libraries: =`
2. then for each of these lines only take the part that comes after the `=` sign
3. split that part by `:` and iterate over the pieces

I think you could refactor this to make it a bit more readable by using `filter_map()` instead of filter to also remove the `libraries: =` prefix at the same time. Something like:

```rust
.lines()
.filter_map(|line| if line.starts_with("libraries: =") {
    Some(line.replacen("libraries: =", "", 1))
} else {
    None
})
.collect();
```

and then keep the

```rust
for line in lines {
    for path in line.split(':') {
        ...
    }
}
```

It's not necessarily shorter (actually, I think it might be longer) but I think it's a bit more readable at first glance.